JASP-Desktop/widgets/rowcontrols.cpp

```diff
@@ -46,22 +46,33 @@ void RowControls::init(int row, const QString& key)
 
 		QObject* obj = comp->create(context);
 
-		if (obj)	_rowObjects.push_back(QVariant::fromValue(obj));
-		else		Log::log() << "Could not create control in ListView " << listView->name() << std::endl;
+		if (obj)
+		{
+			_contextMap[obj] = context;
+			_rowObjects.push_back(QVariant::fromValue(obj));
+		}
+		else
+			Log::log() << "Could not create control in ListView " << listView->name() << std::endl;
 
 		col++;
 	}
 }
 
 void RowControls::setContext(int row, const QString &key)
 {
-	for (JASPControlWrapper* controlWrapper : _rowJASPWrapperMap.values())
+	QMapIterator<QObject*, QQmlContext*> it(_contextMap);
```

Comment:
Why not use: `for(auto & itemCtxt : _contextMap)`
And then refer to item as `itemCtxt.first` and to context as `itemCtxt.second`.
That would make this all a but less messy.

Reply:
It is QMap, not std::map... your code does not work with QMap.

Reply:
No, then why not use std::map instead as well?

Reply:
OK